packages/panzer/dof-mgr/src/Panzer_DOFManager_impl.hpp

```diff
@@ -569,17 +569,22 @@ void DOFManager<LO,GO>::buildGlobalUnknowns(const Teuchos::RCP<const FieldPatter
     HashTable isOwned, remainingOwned;
 
     // owned_ is made up of owned_ids.: This doesn't work for high order
-    Teuchos::ArrayRCP<const GO> nvals = non_overlap_mv->get1dView();
-    Teuchos::ArrayRCP<const GO> tagged_vals = tagged_non_overlap_mv->get1dView();
+    non_overlap_mv->sync_host();
+    tagged_non_overlap_mv->sync_host();
+    PHX::Device::fence();
+    auto nvals = non_overlap_mv->template getLocalView<PHX::Device>();
+    auto tagged_vals = tagged_non_overlap_mv->template getLocalView<PHX::Device>();
     TEUCHOS_ASSERT(nvals.size()==tagged_vals.size());
-    for (int j = 0; j < nvals.size(); ++j) {
-      if (nvals[j] != -1) {
-        for(GO offset=0;offset<tagged_vals[j];offset++)
-          isOwned.insert(nvals[j]+offset);
-      }
-      else {
-        // sanity check
-        TEUCHOS_ASSERT(tagged_vals[j]==0)
+    for (size_t i = 0; i < nvals.extent(1); ++i) {
```

Comment:
I think Kokkos::View has a `size_type` typedef that would be the best thing to use here.

Reply:
Kokkos used to map the return type to a NODE dependency but when they moved to extent() it got hard coded to size_t. Here's the code from the View object:
```
  template< typename iType >
  KOKKOS_INLINE_FUNCTION constexpr
  typename std::enable_if< std::is_integral<iType>::value , size_t >::type
  extent( const iType & r ) const
    { return m_map.extent(r); }
```

Reply:
Ah, OK, sorry about that!